webapp/src/pages/Settings.tsx

```diff
@@ -207,6 +217,40 @@ const Settings: React.FC = () => {
     Partial<AzimuthConfig>
   >({});
 
+  React.useEffect(() => {
+    if (config && defaultConfig && defaultConfig.language !== config.language) {
+      setPartialConfig({
+        language: defaultConfig.language,
+        syntax: {
+          ...(partialConfig.syntax ?? config.syntax),
+          spacy_model: defaultConfig.syntax.spacy_model,
+          subj_tags: defaultConfig.syntax.subj_tags,
+          obj_tags: defaultConfig.syntax.obj_tags,
+        },
+        similarity: {
```

Comment:
Changing language should not change the checkbox for enabling/disabling similarity. I think you have to do something like this to preserve the `null`:
```suggestion
        similarity: partialConfig.similarity && {
```

Reply:
Ohh right! Good catch!